packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -181,23 +159,21 @@ export const transformPlotlyJsonToScatterChartProps = (
   }
 };
 
-export const transformPlotlyJsonToHorizontalBarWithAxisProps = (jsonObj: any): IHorizontalBarChartWithAxisProps => {
-  const getColor = UseColorMapping();
+export const transformPlotlyJsonToHorizontalBarWithAxisProps = (
+  jsonObj: any,
+  getColor: (colorString: string) => string,
+): IHorizontalBarChartWithAxisProps => {
   const { data, layout } = jsonObj;
 
   const chartData: IHorizontalBarChartWithAxisDataPoint[] = data
     .map((series: any, index: number) => {
       return series.y.map((yValue: string, i: number) => {
-        const { color, opacity } = getColor(series.marker?.color, i);
+        const color = Array.isArray(series.marker?.color) ? getColor(series.marker?.color[i]) : getColor('');
```

Comment:
use the isArrayOrTypedArray utility defined below.

Reply:
done